serverless/pytorch/facebookresearch/detectron2/retinanet/nuclio/main.py

```diff
@@ -0,0 +1,67 @@
+import json
+import base64
+import io
+from PIL import Image
+import yaml
+
+import torch
+from detectron2.config import get_cfg
+from detectron2.data.detection_utils import convert_PIL_to_numpy
+from detectron2.engine.defaults import DefaultPredictor
+from detectron2.data.datasets.builtin_meta import COCO_CATEGORIES
+
+CONFIG_FILE = "detectron2/configs/COCO-Detection/retinanet_R_101_FPN_3x.yaml"
+CONFIG_OPTS = ["MODEL.WEIGHTS", "model_final_971ab9.pkl"]
+CONFIDENCE_THRESHOLD = 0.5
+
+def init_context(context):
+    context.logger.info("Init context...  0%")
+    if torch.cuda.is_available():
+        functionconfig = yaml.safe_load(open("/opt/nuclio/function-gpu.yaml"))
+        CONFIG_OPTS.extend(['MODEL.DEVICE', 'cuda'])
+    else:
+        functionconfig = yaml.safe_load(open("/opt/nuclio/function.yaml"))
+        CONFIG_OPTS.extend(['MODEL.DEVICE', 'cpu'])
+
+    cfg = get_cfg()
+    cfg.merge_from_file(CONFIG_FILE)
+    cfg.merge_from_list(CONFIG_OPTS)
+    cfg.MODEL.RETINANET.SCORE_THRESH_TEST = CONFIDENCE_THRESHOLD
+    cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = CONFIDENCE_THRESHOLD
+    cfg.MODEL.PANOPTIC_FPN.COMBINE.INSTANCES_CONFIDENCE_THRESH = CONFIDENCE_THRESHOLD
+    cfg.freeze()
+    predictor = DefaultPredictor(cfg)
+
+    setattr(context.user_data, 'model_handler', predictor)
+    labels_spec = functionconfig['metadata']['annotations']['spec']
+    labels = {item['id']: item['name'] for item in json.loads(labels_spec)}
+    setattr(context.user_data, "labels", labels)
+
+    context.logger.info("Init context...100%")
+
+def handler(context, event):
+    context.logger.info("Run retinanet-R101 model")
+    data = event.body
+    buf = io.BytesIO(base64.b64decode(data["image"].encode('utf-8')))
```

Comment:
```suggestion
    buf = io.BytesIO(base64.b64decode(data["image"]))
```

It should be unnecessary to encode the string, as `b64decode` accepts character strings.